### PR TITLE
Updating Canary genesis block with TransmissionID checksum - v0.4.0

### PR DESCRIPTION
**Pull request by zosorock**
## Motivation

Please see below for improvements included in this release:
[[Fix] Add checksum to TransmissionID](https://github.com/AleoNet/snarkOS/pull/3367)

## Related PRs
The sister PR is https://github.com/AleoNet/snarkVM/pull/2522
https://github.com/AleoNet/snarkVM/pull/2521
[AleoNet/snarkOS#3367](https://github.com/AleoNet/snarkOS/pull/3367)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ default-features = false
 [workspace.dependencies.snarkvm]
 #path = "../snarkVM"
 git = "https://github.com/AleoNet/snarkVM.git"
-rev = "68f4f31" # If this is updated, the rev in `node/rest/Cargo.toml` must be updated as well.
+rev = "3d8b912" # If this is updated, the rev in `node/rest/Cargo.toml` must be updated as well.
```
Comment (raychu86) [inline Cargo.toml:50]: You need to update the rev in `node/rest/Cargo.toml` as well

**Comment by Meshiest** [comment]
are there supposed to be Cargo.lock changes too?

**Comment by vicsn** [comment]
> are there supposed to be Cargo.lock changes too?

Yes @zosorock after you updated the rev in `Cargo.toml`, the `Cargo.lock` will always change a couple seconds later automatically. Can you `git add` and `commit` it as well? It doesn't affect any logic.

In the past we've ignored this detail because we were in a timecrunch but it should be done. I noted it now too in the WIP `snarkVM: Contribution Guide` document.

**Comment by vicsn** [comment]
The `handshake_responder_side_invalid_challenge_response` test is failing because it was not designed to handle version changes. Find enclosed a diff which fixes it, you can apply using `git apply diff.txt`.

[diff.txt](https://github.com/user-attachments/files/16440640/diff.txt)

[CI succeeding](https://app.circleci.com/pipelines/github/ProvableHQ/snarkOS/13598/workflows/d1372bda-8dda-4991-9790-569b78d420f8) for ProvableHQ mirror https://github.com/ProvableHQ/snarkOS/tree/canary/v0.4.0

**Review by snarkworld — APPROVED** [review]
LGTM, the `Cargo.lock` and `diff.txt` update will be handled in separate PR.

**Pull request merged**